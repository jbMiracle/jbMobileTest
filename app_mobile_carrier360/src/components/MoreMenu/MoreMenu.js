import React from 'react';
import { Text, View, FlatList, Image } from 'react-native';
import { Menu, MenuOptions, MenuOption, MenuTrigger, renderers } from 'react-native-popup-menu';
import { menuOptionsStyle } from '../C360Header/C360Header.style';
import C360Colors from '../../themes/360Colors';
import C360Icon from '../C360Icon/C360Icon';
import strings from '../../services/localization';

const demoProfileAvatar = require('../../assets/images/profileAvatar.jpeg');

const { Popover } = renderers;
const menuTextOptions = { scope: 'MoreMenu' };

const MoreMenu = ({ menuList, style }) => {
  const {
    headerIcon,
    profileMenuView,
    mainText,
    subText,
    flexColumn,
    popoverProps,
    menuTriggerStyles,
  } = style;
  const mainMenu = menuOptionsStyle({ type: 'main' });

  return (
    <Menu renderer={ Popover } rendererProps={ popoverProps } >
      <MenuTrigger customStyles={ menuTriggerStyles }>
        <C360Icon
          style={ headerIcon }
          name='jbh-Menu_Card'
          size={ 23 }
          color={ C360Colors.White }
        />
      </MenuTrigger>
      <MenuOptions customStyles={ mainMenu }>
        <View style={ profileMenuView }>
          { renderProfileAvatar(style) }
          <View style={ flexColumn }>
            <Text style={ mainText }>David Hasselhoff</Text>
            <Text style={ subText }>Driver</Text>
            <Text style={ subText }>ABC Trucking</Text>
          </View>
        </View>
        <FlatList
          data={ menuList }
          bounces={ false }
          renderItem={ renderMenuItem(style) }
        />
      </MenuOptions>
    </Menu>
  );
};

function renderMenuItem(style) {
  const { marginRight25 } = style;
  const mainMenu = menuOptionsStyle({ type: 'main' });

  return ({ item }) => (
    <MenuOption value={ item.key }>
      <C360Icon style={ marginRight25 } name={ item.icon } size={ 15 } color={ C360Colors.Grey900 } />
      <Text style={ mainMenu.optionText }>{strings(item.key, menuTextOptions)}</Text>
    </MenuOption>
  );
}

function renderProfileAvatar(style) {
  // for now, avatarExists is always true for demo purposes, until we get actual accounts in the app
  const avatarExists = true;
  const avatar = avatarExists
    ? demoProfileAvatar
    : demoProfileAvatar; // here, we will have the placeholder image

  return (
    <Image
      resizeMode='cover'
      style={ style.profileAvatar }
      source={ avatar }
    />
  );
}

export default MoreMenu;
