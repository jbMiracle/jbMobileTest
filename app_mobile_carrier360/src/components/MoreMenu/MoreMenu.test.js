import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MoreMenu from './MoreMenu';

it('can render a menu list component', () => {
  const renderer = new ShallowRenderer();

  const props = {
    menuList: [
      { key: 'Profile', icon: 'notebook' },
      { key: 'Manage Users', icon: 'people' },
      { key: 'Feedback', icon: 'envelope' },
      { key: 'Sign Out', icon: 'logout' },
    ],
    style: {
      headerIcon: {},
      profileMenuView: '',
      mainText: '',
      subText: '',
      flexColumn: '',
      popoverProps: {},
      menuTriggerStyles: {},
    },
  };

  renderer.render(<MoreMenu { ...props } />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
