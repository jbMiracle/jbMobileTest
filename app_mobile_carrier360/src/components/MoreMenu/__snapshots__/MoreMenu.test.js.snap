// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render a menu list component 1`] = `
<Menu
  onBackdropPress={[Function]}
  onClose={[Function]}
  onOpen={[Function]}
  onSelect={[Function]}
  renderer={[Function]}
  rendererProps={Object {}}
>
  <MenuTrigger
    customStyles={Object {}}
    disabled={false}
  >
    <Icon
      allowFontScaling={false}
      color="#FFFFFF"
      name="jbh-Menu_Card"
      size={23}
      style={Object {}}
    />
  </MenuTrigger>
  <MenuOptions
    customStyles={
      Object {
        "optionText": Object {
          "color": "#444444",
          "fontSize": 12,
        },
        "optionTouchable": Object {
          "activeOpacity": 0.3,
          "underlayColor": "transparent",
        },
        "optionWrapper": Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "margin": 5,
        },
        "optionsContainer": Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 7,
        },
        "optionsWrapper": Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 7,
          "paddingHorizontal": 20,
          "paddingVertical": 10,
          "width": 449.10179640718565,
        },
      }
    }
  >
    <View
      style=""
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../src/assets/images/profileAvatar.jpeg",
          }
        }
        style={undefined}
      />
      <View
        style=""
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style=""
        >
          David Hasselhoff
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style=""
        >
          Driver
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style=""
        >
          ABC Trucking
        </Text>
      </View>
    </View>
    <FlatList
      bounces={false}
      data={
        Array [
          Object {
            "icon": "notebook",
            "key": "Profile",
          },
          Object {
            "icon": "people",
            "key": "Manage Users",
          },
          Object {
            "icon": "envelope",
            "key": "Feedback",
          },
          Object {
            "icon": "logout",
            "key": "Sign Out",
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </MenuOptions>
</Menu>
`;
